From 616e75ea3717757aec4495ac35c87cf41f519cfa Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 22 Mar 2021 22:54:28 +0800
Subject: [PATCH 40/79] desktop-shell: Use real size for non-image background

Currently we are faking 1x1 size for non-image background, which would
causing incorrect opaque region.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 clients/desktop-shell.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/clients/desktop-shell.c b/clients/desktop-shell.c
index 8711399..6cf21da 100644
--- a/clients/desktop-shell.c
+++ b/clients/desktop-shell.c
@@ -870,12 +870,6 @@ background_configure(void *data,
 		return;
 	}
 
-	if (!background->image && background->color) {
-		widget_set_viewport_destination(background->widget, width, height);
-		width = 1;
-		height = 1;
-	}
-
 	widget_schedule_resize(background->widget, width, height);
 }
 
-- 
2.20.1

